atlasdb-client/src/main/java/com/palantir/atlasdb/transaction/impl/CachingTransaction.java

```diff
@@ -104,30 +105,38 @@ public Transaction delegate() {
                 }
             }
             SortedMap<byte[], RowResult<byte[]>> results = super.getRows(tableRef, toLoad, columnSelection);
-            cacheLoadedRows(colCache, results.values());
+            cacheLoadedRows(tableRef, results.values());
             inCache.putAll(results);
             return inCache.build();
         }
     }
 
-    private void cacheLoadedRows(ConcurrentMap<Cell, byte[]> colCache,
+    private byte[] getCachedCellIfPresent(TableReference tableRef, Cell cell) {
+        return cellCache.getIfPresent(Pair.create(tableRef.getQualifiedName(), cell));
+    }
+
+    private void cacheLoadedCell(TableReference tableRef, Cell cell, byte[] value) {
```

Comment:
For readability, please put this method below its usages. `getCachedCellIfPresent` could even follow it down - they're both at the same level of abstraction.